compat/osdetect.h

```diff
@@ -66,7 +66,6 @@
 
 #elif defined(__MINGW32__) || defined(__MINGW__)
 #define _SQUID_MINGW_ 1
-#define _SQUID_WINDOWS_ 1
```

Comment:
All libcompat code that is currently inside `_SQUID_WINDOWS_` guards is **theoretically** dead code. ie it has not (yet) had a proven need to exist.

BUT, I do not dare drop it completely at this time since there are many parts of Squid still to be fixed which might prove a need and involve corrections to those wrappers.

Reply:
The connection is that it is dead code either way. Code in src/ is out of scope for this PR.

Reply:
> In summary, please make SQUID_WINDOWS and SQUID_MINGW guard usage consistent with their new mutually exclusive definition.

There are two places in the code which do not (yet) meet that criteria.
 * code in src/ which is out-of-scope for this PR. These are being resolved in followup PRs.
 * code in the `mswindows.{h,cc}` files. That is being left **for now** for the same reason all code inside other _SQUID_WINDOWS_ wrappers are being left as-is. Changing it requires the MSVC compiler to verify the change correctness.

Reply:
You appear to have missed my point entirely. The requested change has far-reaching implications that I refuse to bring in-scope to this PR.
As much as you dislike the macro naming and usage it is the status-quo we have today. Please withdraw this request to change practice/policy.

Reply:
> The requested change has far-reaching implications

What far-reaching implications do you see?

> [the macro naming and usage] is the status-quo we have today

This PR changes the status quo:

* Before this PR, MinGW builds were a flavor or sub-category of Windows builds. MinGW builds used general Windows code (`_SQUID_WINDOWS_`) plus some MinGW-specific code.

* After this PR: MinGW builds are first-class or stand-alone builds like Linux or FreeBSD builds. They no longer use general Windows code (`_SQUID_WINDOWS_`).

The value of the `_SQUID_MINGW_` macro (i.e. 1) stays the same. The name of the macro guarding MinGW build is not of primary importance (to me) and can stay `_SQUID_MINGW_` if we want it to stay the same. However, this PR changes the semantics of what "MinGW build" means -- what code is enabled, what code is not enabled, what macros are defined, what macros are not defined, etc. Once this request is addressed, PR code will become consistent with that change in MinGW build semantics. Addressing this change request is important and straightforward. I have suggested specific ways to address it.

> Please withdraw this request to change practice/policy.

This change request does not request any practice or policy change. It only requests to remove an inconsistency introduced by PR changes.

Reply:
@yadij, please see compat/os/mswindows.h in master for an example where `_SQUID_MINGW_` is still used inside `_SQUID_WINDOWS_`.

```
upstream/master:compat/os/mswindows.h-17-#if _SQUID_WINDOWS_
upstream/master:compat/os/mswindows.h-18-
upstream/master:compat/os/mswindows.h-19-/****************************************************************************
upstream/master:compat/os/mswindows.h-20- *--------------------------------------------------------------------------*
upstream/master:compat/os/mswindows.h-21- * DO *NOT* MAKE ANY CHANGES below here unless you know what you're doing...*
upstream/master:compat/os/mswindows.h-22- *--------------------------------------------------------------------------*
upstream/master:compat/os/mswindows.h-23- ****************************************************************************/
upstream/master:compat/os/mswindows.h-24-
upstream/master:compat/os/mswindows.h-25-/* we target Windows XP and later - some API are missing otherwise */
upstream/master:compat/os/mswindows.h:26:#if _SQUID_MINGW_
```

Same for the current PR code.

I did not check for other occurrences. Please do.

Reply:
Would it be worth, in the future, to further unravel by declaring:

_SQUID_WINDOWS_ is code targeting the windows platform
_SQUID_MINGW_ is only code targetining mingw
_SQUID_MSVC_ or something similar targets specific oddities of the MSVC compiler

?

Reply:
> Amos: FTR, cleared up the misunderstanding during a dev Zoom meeting. The issue of _SQUID_MINGW_ use in code not touched by this PR is resolved.

Just to avoid a misunderstanding: This change request is still not addressed. During the meeting, you mentioned that the original problem went away because other PRs removed problematic "`_SQUID_MINGW_` inside `_SQUID_WINDOWS_`" code. I could not quickly check that assertion during the meeting, so I said something along the lines of "this change request is addressed provided that assertion is true (i.e. the problematic code is gone)".

After the meeting, I have checked that assertion: The problem is still there. After the meeting, I have [quoted](https://github.com/squid-cache/squid/pull/1341#discussion_r1243334118) some of the code that introduces the inconsistency that this change request is devoted to. I believe there are more such cases inside compat/os/mswindows.h (at least). If necessary, I can identify all such problematic places. I am happy to clarify anything that needs clarification, but the ball is on your side.

> Francesco: `_SQUID_WINDOWS_` is code targeting the windows platform

AFAICT, after this PR: `_SQUID_WINDOWS_` is code targeting the windows platform _with the exception_ of MinGW and CygWin builds. MinGW used to be included in `_SQUID_WINDOWS_`. This PR proposes to make the two mutually exclusive. There are no objections to that proposal, but the current PR code does not yet make the two mutually exclusive (from a human code reader point of view) -- we still have some `#if _SQUID_WINDOWS_ ... #if _SQUID_MINGW_` sections. This change request is about removing that inconsistency. There are several ways to remove it; some of the proposed solutions do not require a lot of work or a lot of changes.

Reply:
Oh, those bits dropped in [f660d38](https://github.com/squid-cache/squid/pull/1341/commits/f660d3833d2f15404ee1a114b944598b93c44ebe). I was checking the src/ etc directories earlier.